UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs

```diff
@@ -423,8 +482,9 @@ private void SetImageSprite(Sprite value)
 			{
 				image.enabled = false;
 			}
-
 		}
+
+		OnSpriteChanged?.Invoke(value);
```

Comment:
how much of a performance impact is this?

Reply:
This event invokes for each frame in object animation (gap between frames usually 0.2-0.4 second). For non-animated items or objects it is just one null check. It's nothing.
For animated items this event just replace sprite in UI Image class. Performance here is even better than before. In old approach custom script used to create coroutine or even Update cycle to manually refresh UI slot.

Keep in mind that this is Client side code only. From my test I can't see any impact on performance.